py-polars/tests/unit/functions/test_lit.py

```diff
@@ -99,6 +100,42 @@ def test_lit_unsupported_type() -> None:
         pl.lit(pl.LazyFrame({"a": [1, 2, 3]}))
 
 
+def test_lit_enum_input_16668() -> None:
+    # https://github.com/pola-rs/polars/issues/16668
+
+    class State(str, enum.Enum):
+        VIC = "victoria"
+        NSW = "new south wales"
+
+    value = State.VIC
+
+    result = pl.lit(value)
+    assert pl.select(result).dtypes[0] == pl.Enum(["victoria", "new south wales"])
+    assert pl.select(result).item() == "victoria"
+
+    result = pl.lit(value, dtype=pl.String)
+    assert pl.select(result).dtypes[0] == pl.String
+    assert pl.select(result).item() == "victoria"
+
```

Comment:
Might be good to check if the original issue also is accounted for.
```suggestion
    assert pl.select(result).item() == State.VIC
```

Reply:
`.item()` is going to return a flat string, not an Enum. Not sure what Python does with Enums and string equality, but since `str(State.Vic) == "State.Vic"` bit us previously I'm not going to make any assumptions about how Python handles that.

Reply:
@stinodego, If you view my original issue I made, python handles State.VIC as a string, I gave several examples there.